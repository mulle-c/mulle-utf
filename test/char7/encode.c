#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>
#include <string.h>



int  main()
{
   int         c;
   int         i;
   int         d;
   uint32_t    code;
   char        buf[ 7];
   size_t      size;

   /**/
   /**/
   /**/

   code = mulle_char7_encode32( ".u", 2);
   size = mulle_char7_decode32( code, buf, 2);
   buf[ size] = 0;

   printf( "%llx\n", (long long) code);

   for( i = 0; i < 2; i++)
      putchar( mulle_char7_get( code, i));
   printf( "\n");

   /**/
   /**/
   /**/

   memset( buf, 'X', sizeof( buf));

   code = mulle_char7_encode32( "LMPR", 4);
   size = mulle_char7_decode32( code, buf, 4);

   if( buf[ 5] != 'X')
   {
      printf( "encode decode overflow\n");
      abort();
   }
   buf[ size] = 0;

   if( strncmp( buf, "LMPR", 4))
   {
      printf( "encode decode failed\n");
      abort();
   }

   printf( "%llx\n", (long long) code);

   for( i = 0; i < 4; i++)
      putchar( mulle_char7_get( code, i));

   printf( "\n");

   return( 0);
}

