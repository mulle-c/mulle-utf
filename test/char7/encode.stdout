3aae
.u
a5426cc
LMPR
